src/include/platform/internal/GenericConfigurationManagerImpl.ipp

```diff
@@ -351,16 +351,52 @@ CHIP_ERROR GenericConfigurationManagerImpl<ConfigClass>::GetSecondaryPairingHint
 template <class ConfigClass>
 CHIP_ERROR GenericConfigurationManagerImpl<ConfigClass>::GetSoftwareVersionString(char * buf, size_t bufSize)
 {
-    ReturnErrorCodeIf(bufSize < sizeof(CHIP_DEVICE_CONFIG_DEVICE_SOFTWARE_VERSION_STRING), CHIP_ERROR_BUFFER_TOO_SMALL);
-    strcpy(buf, CHIP_DEVICE_CONFIG_DEVICE_SOFTWARE_VERSION_STRING);
-    return CHIP_NO_ERROR;
+    ChipError err       = CHIP_NO_ERROR;
+    size_t softwareVersionStringLen = 0; // without counting null-terminator
+
+    err = ReadConfigValueStr(ConfigClass::kConfigKey_SoftwareVersionString, buf, bufSize, softwareVersionStringLen);
+#ifdef CHIP_DEVICE_CONFIG_DEVICE_SOFTWARE_VERSION_STRING
```

Comment:
This should always be defined, no?

Reply:
looks like its not, per what is defined in the CHIPDeviceConfig definitions

Reply:
How could that be, if you're including CHIPDeviceConfig.h? But this file is _not including it_, so you are not seeing the right definition for things, no?  That's the bug, and the ifdef is hiding it.

Reply:
fixed